### PR TITLE
Refactor List Item dom structure

### LINKED ISSUE #15186: Typography: marked text not displayed correctly inside list item
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/8z503zj7m9)

### Steps to reproduce
Use the new `Typography` component to display marked text inside a list item.

### What is expected?
The text inside the list item is expected to look the same as without the enclosing list item (i. e. like the first line in the MWE).

### What is actually happening?
The marked text inside the list item has no space between itself and the surrounding words.

| Environment | Info |
|---|---|
| antd | 3.14.1 |
| React | 16.8.3 |
| System | Windows 10 Pro |
| Browser | Google Chrome 72.0.3626.119 (64-bit) |

---
The issues seems to stem from the `display: flex` of the enclosing `List.Item`.

### PR DESCRIPTION
**Pull request by afc163**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

#15186 told a bug that default flex layout will cut all spaces in plain string of children.

[Edit on CodeSandbox](https://codesandbox.io/s/8z503zj7m9)

image

### 💡 Solution

After a deep digging, I decide to refactor whole list item dom structure which is a mess. And resolve #15186 via check it there any string element under children.

https://github.com/ant-design/ant-design/pull/15210/files#diff-9cef4c52eb35ccf96cbdc9432372ebb5R73

### 📝 Changelog description

> Describe changes from user side, and list all potential break changes or other risks.

1. English description

- ⚡️ Refactor and simplify List Item dom structure, and fix spaces lost inside List Item. #15210

2. Chinese description (optional)

- ⚡️ 重构并简化了 List Item 的 dom 结构，并且修复了 Item 中内容空格丢失的问题。#15210

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/list/Item.tsx [hunk 1]
```diff
@@ -81,52 +92,38 @@ export default class Item extends React.Component<ListItemProps, any> {
...
-      actionsContent = (
-        <ul className={`${prefixCls}-item-action`}>
-          {actions.map((action, i) => actionsContentItem(action, i))}
-        </ul>
-      );
-    }
-
-    const extraContent = (
-      <div className={`${prefixCls}-item-extra-wrap`}>
-        <div className={`${prefixCls}-item-main`}>
```
Comment (afc163) [inline components/list/Item.tsx]: 简化了这些多余的 dom 结构。

File: components/list/Item.tsx [hunk 2]
```diff
@@ -81,52 +92,38 @@ export default class Item extends React.Component<ListItemProps, any> {
...
-
-    React.Children.forEach(children, (element: React.ReactElement<any>) => {
-      if (element && element.type && element.type === Meta) {
-        metaContent.push(element);
-      } else {
-        otherContent.push(element);
-      }
-    });
-
-    const contentClassString = classNames(`${prefixCls}-item-content`, {
```
Comment (afc163) [inline components/list/Item.tsx]: 简化了这些多余的 dom 结构。

File: components/list/Item.tsx [hunk 3]
```diff
@@ -70,6 +70,17 @@ export default class Item extends React.Component<ListItemProps, any> {
 
   context: any;
 
+  isItemContainsTextNode() {
```
Comment (zombieJ) [inline components/list/Item.tsx]: 感觉会覆盖不全：
```<span>I'm</span><strong>{' also '}</strong><span>text</span>```

**Comment by AENeuro** [comment]
I'm not sure but does this commit break the itemLayout property?

**Comment by zombieJ** [comment]
hi @AENeuro , please feel free to open an issue if it breaks something.

**Pull request merged**